test/radio/radio-controller-test.ts

```diff
@@ -0,0 +1,81 @@
+import test from 'ava';
+import * as dgram from 'dgram';
+import {RadioController, UDPScanConfig} from '../../src/radio/radio-controller';
+import {UdpResponse} from '../../src';
+
+const UDP_DUMMY_BUFFER_1 = Buffer.from('test UDP buffer');
+const UDP_DUMMY_BUFFER_2 = Buffer.from('foo bar lorem ipsum');
+
+/**
+ * Class to manage a local UDP server
+ */
+class UdpServer {
+  private server: dgram.Socket | undefined;
+  public startServer(listenPort = 3112) {
+    this.server = dgram.createSocket('udp4');
+    this.server.on('message', (message: Buffer, rinfo) => {
+      this.server!.send(UDP_DUMMY_BUFFER_1, rinfo.port, rinfo.address);
```

Comment:
Do we need to assert that the server is not undefined? What would happen if it is undefined?

Reply:
As far as I can tell it is impossible that `this.server` will be undefined at this point, since it's a private member that is definitely set in the first line of `startServer`, and never set back to `undefined`.